org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintGlobalConfiguration.java

```diff
@@ -266,7 +271,7 @@ private static Set<RuleConfig> deserializeRulesJson(String json) {
 
   public static void saveRulesConfig(Collection<RuleConfig> rules) {
     var json = serializeRulesJson(rules);
-    setPreferenceString(PREF_RULES_CONFIG, json);
+    setPreferenceString(getApplicationLevelPreferenceNode(), PREF_RULES_CONFIG, json);
```

Comment:
Isn't it a breaking change? People having previously customized their rule config (stored in the workspace scope) will lose their config.
You should maybe implement a migration when reading (try to read from the old location, and if present, copy the settings to the new location).

Reply:
Ah, you are right, sorry. Then LGTM!